### PR TITLE
Adding solaris recipes

### PR DESCRIPTION
**Pull request by nrweir**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/solaris-gpu/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    - rtree >=0.8.3
+    - scikit-image >=0.14.0
+    - scipy >=1.2.0
+    - shapely >=1.6.4
+    - tensorflow-gpu >1.10, <=1.13.1
+    - cuda90
+    - pytorch >=1.0.1
+    - torchvision >=0.3.0
+    - tqdm >=4.32.2
+    - urllib3 ==1.24.3
```
Comment (beckermr) [inline recipes/solaris-gpu/meta.yaml]: These pins are excessive. When shipping a library, only the minimally absolutely necessary pins should be used. Otherwise, your package won't be able to coexist in envs with others.

File: conda-forge.yml [hunk 2]
```diff
@@ -1,2 +1,9 @@
+min_py_ver: '35'
 max_py_ver: '37'
-max_r_ver: '35'
\ No newline at end of file
+max_r_ver: '35'
+
+channels:
```
Comment (beckermr) [inline conda-forge.yml]: I am not sure if this section should be here. I defer to other experts on @conda-forge/staged-recipes. Generally, all dependencies should be in conda-forge.

File: recipes/solaris-gpu/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    - mask_to_polygons = solaris.bin.mask_to_polygons:main
+    - spacenet_eval = solaris.bin.spacenet_eval:main
+    - solaris_run_ml = solaris.bin.solaris_run_ml:main
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed --no-cache-dir -vv "
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+  host:
+    - {{ compiler('c') }}
```
Comment (beckermr) [inline recipes/solaris-gpu/meta.yaml]: Compilers do not go in the host section.

File: recipes/solaris/LICENSE.txt [hunk 4]
```diff
@@ -0,0 +1,201 @@
+Apache License
```
Comment (beckermr) [inline recipes/solaris/LICENSE.txt]: It is fine to include the license here. However, you should make an upstream issue/commit to make sure the license is included in the pypi sdist.

Comment (nrweir) [reply recipes/solaris/LICENSE.txt]: Thanks - it is.

File: conda-forge.yml (hunk 2 above)
Comment (nrweir) [reply conda-forge.yml]: I know that's generally the case, but unfortunately it's currently impossible for `pytorch`, which is why I added that channel (per discussions in gitter)

File: recipes/solaris-gpu/meta.yaml (hunk 1 above)
Comment (nrweir) [reply recipes/solaris-gpu/meta.yaml]: I can loosen where it won't interfere with functionality, thanks

File: recipes/solaris/LICENSE.txt (hunk 4 above)
Comment (beckermr) [reply recipes/solaris/LICENSE.txt]: Cool. If the license file is upstream then there is no reason to ship it here and we can delete it.

File: recipes/solaris-gpu/meta.yaml (hunk 3 above)
Comment (nrweir) [reply recipes/solaris-gpu/meta.yaml]: Thanks - fixed in the new commit.

**Comment by jjhelmus** [comment]
The conda-forge Tensorflow pacakge is not built from source and does not support CentOS 6, see conda-forge/tensorflow-feedstock#6.  Any downstream packages which try to import tensorflow will fail in the test environment.

**Comment by nrweir** [comment]
@jjhelmus thanks for the heads up - should we not worry about this test failure at the moment then?

File: conda-forge.yml (hunk 2 above)
Comment (isuruf) [reply conda-forge.yml]: Which gitter conversation was this? Can you link to it? 
We have never allowed other channels in conda-forge.

**Comment by nrweir** [comment]
@beckermr I made changes per your request in https://github.com/conda-forge/staged-recipes/pull/8837/commits/b1d8ba1a6a0002c041ea60ecc08a99f2fde89f83

**Comment by beckermr** [comment]
Wow I did not realize where we were with tensorflow and pytorch. Software is hard. :/

File: conda-forge.yml (hunk 2 above)
Comment (nrweir) [reply conda-forge.yml]: Sure. I don't know how to link to the specific convo, but it's pasted below. Timestamp is Jul 08 2019 17:13.

adbeda:
> question: 
> how can I specify a pinning dependency from a different channel?
> 
> I'm creating a recipe for  snorkel (https://github.com/HazyResearch/snorkel) and this package requires `pytorch==1.1.0` which is doesn't exist in conda-forge channel.  pytorch is available on conda but through its own channel. (i.e. https://anaconda.org/pytorch/pytorch).  
> 
> github issues I read said that its not possible to use different channels at this point but is there a better solution other than attempting to bring pytorch to conda-forge first before working on snorkel?

ascopatz:
> You can use other channels if you set the channels in `conda-forge.yml`: https://conda-forge.org/docs/maintainer/conda_forge_yml.html#channels

> This really isn't encouraged though, because it requires that the user have that channel when they install too

> It is ultimately better to get the deps on conda-forge (or baring that main)

> However, I know what an utter pain pytorch is to build

adbeda:
> ha. thats interesting.  so, what have people done? I mean I can use your suggestions, even when its not encouraged, (to make my life easier) but are there any other acceptable-ish feedstock examples? 

ascopatz:
> You can.  I am not aware of anyone actually doing thing long term
> It is probably OK temporarily

I know it's not recommended/desired, but my impression is that pytorch (and many other GPU-requiring packages) are currently unavailable on conda-forge, despite Herculean efforts on the part of yourself and other maintainers. If there's going to be a hard requirement that any package requiring pytorch and GPUs can't be on conda-forge until those same modules are, there are a significant number of machine learning modules that aren't going to be on conda-forge (until someone with more expertise than a more-or-less pure python data scientist like myself can work on it)

Comment (isuruf) [reply conda-forge.yml]: > If there's going to be a hard requirement that any package requiring pytorch and GPUs can't be on conda-forge until those same modules are, there are a significant number of machine learning modules that aren't going to be on conda-forge (until someone with more expertise than a more-or-less pure python data scientist like myself can work on it)

It is a hard requirement. We don't have control over other channels and we work closely with defaults which is why defaults is allowed. If you want to lift this requirement, write a CFEP https://github.com/conda-forge/conda-forge-enhancement-proposals and then conda-forge/core can vote on it.

Comment (nrweir) [reply conda-forge.yml]: That's fine, I'll find an alternative way to distribute my software.

Comment (adbeda) [reply conda-forge.yml]: @nrweir Can't users be allowed to install Solaris by itself and then pytorch however they want to get it? While building [Snorkel](https://github.com/conda-forge/snorkel-feedstock/blob/master/recipe/meta.yaml#L18-L32), I removed pytorch from a requirement since the pytorch is available on conda-forge  as a cpu variant.

Comment (nrweir) [reply conda-forge.yml]: Unfortunately pytorch is a dependency and we need it for the `solaris` install.

Comment (adbeda) [reply conda-forge.yml]: that makes sense. I'll be working to bring `label-maker` and `raster-vision` in the next couple of weeks. let me know if there is anything I can with
